shell/models/provisioning.cattle.io.cluster.js

```diff
@@ -891,7 +891,37 @@ export default class ProvCluster extends SteveModel {
   }
 
   get hasError() {
-    return this.status?.conditions?.some((condition) => condition.error === true);
+    // Before we were just checking for this.status?.conditions?.some((condition) => condition.error === true)
+    // but this is wrong as an error might exist but it might not be meaningful in the context of readiness of a cluster
+    // which is what this 'hasError' is used for.
+    // We now check if there's a ready condition after an error, which helps dictate the readiness of a cluster
+    // Based on the findings in https://github.com/rancher/dashboard/issues/10043
+    if (this.status?.conditions && this.status?.conditions.length) {
+      if (this.status?.conditions.some((c) => c.error === true)) {
+        // there's no ready condition and has an error, mark it
+        if (!this.status?.conditions.some((c) => c.type === 'Ready')) {
+          return true;
+        }
+
+        const errorConditions = this.status?.conditions.filter((c) => c.error === true);
+        const readyConditions = this.status?.conditions.filter((c) => c.type === 'Ready');
+
+        const lastErrorCondition = errorConditions.reduce((a, b) => ((a.lastUpdateTime > b.lastUpdateTime) ? a : b));
+        const lastReadyCondition = readyConditions.reduce((a, b) => ((a.lastUpdateTime > b.lastUpdateTime) ? a : b));
+
+        const lastErrorConditionDate = new Date(lastErrorCondition.lastUpdateTime) || null;
+        const lastReadyConditionDate = new Date(lastReadyCondition.lastUpdateTime) || null;
+
+        // fallback is false because something might be wrong with the condition 'lastUpdateTime' prop
```

Comment:
What can be wrong with the lastUpdateTime? Why do we need to convert to dates for the comparison, where as above we used the lastUpdateTime directly for comparison?

Reply:
Just wanted to make sure if the BE sent a malformed ISO date Dashboard wouldn't "crash". With the simplified logic based on the previous comment, I've updated this part